### DISCUSSION TITLE
How can I extract an item from a json file?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#52965] [Help] [answered]
### Summary

How can I pin point an specific item from an importer json file?

So I have this code below.

```
const data1 = require("./search-recent.json");

const theJson = JSON.stringify(data1);
console.log(theJson);
```

^^ that works and prints the entire json file..

The json file looks more like this:

```
{
  data: [
    {
      edit_history_ids: ["1682018026445520904"],
      id: "16820180264455209041",
      text: "Saludos!",
    },
    {
      edit_history_ids: ["1681750986363224065"],
      id: "1681750986363224065",
      text: "RT …",
    },
  ],
};

```

What if I want to print the id of the second item? `1681750986363224065`  How can I reach it?

I have been trying all sort of combinations and I can't get it correctly.

```
const data1 = require("./search-recent.json");
const theJson = JSON.stringify(data1);
console.log(theJson.data[1].id);
```

^^^ this keeps giving me errors and I can't get to any values.

```
TypeError: Cannot read properties of undefined (reading '1')
    at Object.<anonymous> 
```

Now, if I copy and paste the json file, and put it behind a constant `information`, and do this below, it works.... but I don't want to copy and paste the json... I want to do the require operation and bring the information from a separate file.

`console.log(information.data[1].id);`

^^  This works if I bring the json and put it behind a constant.. which is what I don't want to do.

**Comment by devjiwonchoi** [comment]
If you know the specific idx of the data, try `...information.data[idx]['edit_history_ids']`

```js
console.log(...information.data[1]['edit_history_ids'])

// or

console.log(information.data[1]['edit_history_ids'][0])
```

### Breakdown:

Access the nth value of data by idx, and get its value by key: 'edit_history_ids'.

This will return something like: `[ '1682018026445520904' ]`.

Now you can use spread syntax or `[0]` at the end to extract the value.

**Comment by Elindo586** [comment]
Would you know how to do it WITHOUT having to print the entire json file and put it behind a constant?

I would like to catch the value from  `const data1 = require("./search-recent.json");`

**Reply by devjiwonchoi** [reply]
Assuming that you know which data to extract,

```js
const data = require('./search-recent.json').data.filter(
  (el) => el['text'] === 'Saludos!' // edit this filter by your needs
)
```

is this somewhat you are looking for?

**Comment by icyJoseph** [comment, marked as answer]
I think the issue here has been the call to `JSON.stringify` that you do on data. After you stringify it, you look for a `data` object on the string, which yields `undefined`, and then you do indexing `[1]` on an undefined value, so you crash out.

In Node.js when you require a json file, it gets automatically parsed as a JavaScript object, which you can work with directly. 

To sum up:

```js
const data1 = require("./search-recent.json");
  // const theJson = JSON.stringify(data1); ---> You should't do this, it turns data1 from a JS object, to a string!
console.log(data1.data[1].id);
```

**Reply by Elindo586** [reply]
Of course it worked! 

Thanks!
